uds/translator/data_record/abstract_data_record.py

```
@@ -211,9 +211,9 @@ def children(self, value: Sequence["AbstractDataRecord"]) -> None:
         for child in value:
             if not isinstance(child, AbstractDataRecord):
                 raise ValueError("At least one of the values in the sequence is not a Data Record.")
-            if child.is_reoccurring:
-                raise ValueError("Child Data Record cannot be reoccurring.")
-            children_length += child.length
+            if not child.fixed_total_length:
+                raise ValueError("Child Data Record have to have fixed total length.")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)